### ISSUE TITLE
500 server error with nginx-proxy and docker-compose

### ISSUE THREAD

**Issue by razaborg** [wallabag/docker#57]
Hi,

I'm trying to use wallabag with the  [jwilder/nginx-proxy](https://hub.docker.com/r/jwilder/nginx-proxy/). I can't figure out why the server is always responding with a 500 error.

The proxy seems to works well given that I got the connection logs using the `docker-compose logs`.

````
wallabag       | 2017/03/08 14:19:24 [error] 309#309: *3 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Exception: Could not gather sufficient random data in /var/www/wallabag/vendor/symfony/symfony/src/Symfony/Component/Security/Csrf/TokenGenerator/UriSafeTokenGenerator.php:46
wallabag       | Stack trace:
wallabag       | #0 /var/www/wallabag/vendor/symfony/symfony/src/Symfony/Component/Security/Csrf/TokenGenerator/UriSafeTokenGenerator.php(46): random_bytes(32)
wallabag       | #1 /var/www/wallabag/vendor/symfony/symfony/src/Symfony/Component/Security/Csrf/CsrfTokenManager.php(57): Symfony\Component\Security\Csrf\TokenGenerator\UriSafeTokenGenerator->generateToken()
wallabag       | #2 /var/www/wallabag/vendor/friendsofsymfony/user-bundle/Controller/SecurityController.php(53): Symfony\Component\Security\Csrf\CsrfTokenManager->getToken('authenticate')
wallabag       | #3 [internal function]: FOS\UserBundle\Controller\SecurityController->loginAction(Object(Symfony\Component\HttpFoundation\Request))
wallabag       | #4 /var/www/wallabag/var/cache/prod/classes.php(4651): call_user_func_array(Array, Array)
wallabag       | #5 /var/www/wallabag/var/cache/prod/classes.php(4606): Symfony\C" while reading response header from upstream, client: ::ffff:172.19.0.7, server: _, request: "GET /login HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "wallabag.mysite.tld"
````
Here is my docker-compose : 

````
 
version: '2'

services:
  nginx-proxy:
    image: jwilder/nginx-proxy:alpine
    container_name: nginx-proxy
    ports:
      - "443:443"
    volumes:
      - /srv/seedbox/htpasswd:/etc/nginx/htpasswd
      - /var/run/docker.sock:/tmp/docker.sock:ro
      - /srv/certs:/etc/nginx/certs

  wallabag:
    image: wallabag/wallabag:latest
    container_name: "wallabag"
    links:
      - mariadb
    environment:
      - MYSQL_ROOT_PASSWORD=myrootpass
      - SYMFONY__ENV__DATABASE_DRIVER=pdo_mysql
      - SYMFONY__ENV__DATABASE_HOST=mariadb
      - SYMFONY__ENV__DATABASE_PORT=3306
      - SYMFONY__ENV__SECRET=mysecret
      - SYMFONY__ENV__DATABASE_NAME=wallabag
      - SYMFONY__ENV__DATABASE_USER=wallabag
      - SYMFONY__ENV__DATABASE_PASSWORD=mypass
      - SYMFONY__ENV__MAILER_HOST=127.0.0.1
      - SYMFONY__ENV__MAILER_USER=~
      - SYMFONY__ENV__MAILER_PASSWORD=~
      - SYMFONY__ENV__FROM_EMAIL=wallabag@mysite.tld
      - VIRTUAL_PORT=80
      - VIRTUAL_HOST=wallabag.mysite.tld
      - VIRTUAL_PROTO=https


  mariadb:
    image: linuxserver/mariadb:latest
    container_name: "mariadb"
    ports:
      - 3306:3306
    volumes:
      - /srv/mariadb:/config
````

Any clues about where the problem could come from ?

**Comment by tcitworld** [comment]
Related to https://github.com/docker-library/php/issues/376 ?

**Comment by razaborg** [comment]
Yes it seems to.

Now I need to downgrade the alpine php7.0.16 package to php7.0.15 to make it work, that's it ?

**Comment by skrzepto** [comment]
Also having same issue with fresh pull of docker image and using sqlite

**Comment by xsteadfastx** [comment]
Strange. Nothing should have changed. We also test fresh pull with travis

**Comment by skrzepto** [comment]
@xsteadfastx  odd that the travis still passes the tests. I was looking at the most recent travis build (https://travis-ci.org/wallabag/docker/jobs/208895684) and it seems to be install the faulting php version that @tcitworld mentions in the logs.

`(52/102) Installing php7 (7.0.16-r0)`

To confirm this issue manually I ran

```
docker run -it --entrypoint=/bin/ash wallabag/wallabag -i
php -r "random_int(0, 1);"
```

output

```
/ # php -r "random_int(0, 1);"
PHP Fatal error:  Uncaught Exception: Could not gather sufficient random data in Command line code:1
Stack trace:
#0 Command line code(1): random_int(0, 1)
#1 {main}
  thrown in Command line code on line 1
```

Someone mentioned in the linked issue that one solution is to `downgrade to 7.0.15 until 7.0.17 releases.`

I think there should be another look at how the test suite is run to verify its actually checking what its supposed to could be a false positive?

**Comment by xsteadfastx** [comment]
Fresh pulled wallabag/wallabag and ran it with docker run --rm -ti -p 80:80 wallabag/wallabag and I could login on a fresh instance. 

I can't reproduce it.

**Comment by xsteadfastx** [comment]
i just deleted all wallabag images on my ubuntu laptop and did a fresh pull. i ran it with `docker run --rm -ti -p 80:80 wallabag/wallabag` and i dont get a 500.

**Comment by razaborg** [comment]
Still the same error for me:
> 2017/03/10 08:59:55 [error] 307#307: *1 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Exception: Could not gather sufficient random data in /var/www/wallabag/vendor/symfony/symfony/src/Symfony/Component/Security/Csrf/TokenGenerator/UriSafeTokenGenerator.php:46

I did exactly the same command as you.
I'm on debian 8, but it's the kernel 3.16.0

**Comment by xsteadfastx** [comment]
Could you do a docker rmi wallabag/wallabag first?

**Comment by razaborg** [comment]
Yep, I just did it.
But still exactly the same issue... blank screen with a 500 error and the weird PHP crypto error

**Comment by xsteadfastx** [comment]
I can't understand that. We should check checksums of the layers. 

Am 10. März 2017 18:48:01 MEZ schrieb Razaborg <notifications@github.com>:

**Comment by xsteadfastx** [comment]
Can it be that it's a problem with the kernel? Did something change with random number creation?

**Comment by xsteadfastx** [comment]
https://bugs.php.net/bug.php?id=74105

**Comment by xsteadfastx** [comment]
could anyone confirm this? maybe with a kernel update or something.

**Comment by nsteinmetz** [comment]
For what I can say, I use default docker-compose for wallabag with Postgres as backend and traefik as reverse proxy and I don't have any issues.

Host system is Debian 8 with 4.5.7 kernel (scaleway vm) and docker 1.13.1

My 2 cents.

**Comment by peetipablo** [comment]
> could anyone confirm this? maybe with a kernel update or something.

I had the same issue.  For now I'm using wallabag:2.1.6 and it's running fine; when the issue is resolved I will update to latest.

**Comment by xsteadfastx** [comment]
so we need to compare kernel versions :) 4.8.0-42 here. because the docker container uses the host kernel this seems to be the reason why it works for some people and for some not.

**Comment by peetipablo** [comment]
3.16.0-4-amd64 - debian 8

**Comment by xsteadfastx** [comment]
it looks like its fixed in php. but i dont know when it gets published.

**Comment by xsteadfastx** [comment]
should be fixed with #65

**Comment by xsteadfastx** [comment]
its fixed now using alpine edge php7 version with the bugfix included for older kernels.
